template/README.md

```diff
@@ -346,6 +346,8 @@ git checkout -
 
 You may copy and paste them, or put them into a custom shell script. You may also customize them for another hosting provider.
 
+Note that if you add a router your app won't work with GitHub Pages becasue it doesn't natively support single page apps with frontend routing (unless it's the ugly hashtag kind). There is a simple workaround for this by adding a `404.html` file with a redirect script, see [Single Page Apps for GitHub Pages](https://github.com/rafrex/spa-github-pages).
```

Comment:
Let’s also avoid words like “simple” and “ugly”. They are very subjective. I would rather a list of workaround, like this:

* You could switch from using HTML5 history API to routing with hashes. The URL will be longer and more verbose (for example, `http://user.github.io/todomvc#todos/42`). For example, if you use React Router, you can switch to `hashHistory` for this effect. [Read more](https://github.com/reactjs/react-router/blob/master/docs/guides/Histories.md#histories) about different history implementations in React Router.

* Alternatively, you can use a trick to teach GitHub Pages to handle 404 by redirecting to your `index.html` page with a special parameter. You would need to add an `404.html` file with the redirection code to the `build` folder before deploying your project, and you’ll need to add code handling that parameter to `index.html`. You can find a detailed explanation of this technique [in this guide](https://github.com/rafrex/spa-github-pages).